src/pages/project/helpers/projectRoleHelpers.ts

```diff
@@ -60,29 +70,181 @@ export const checkIfSameAffiliationOnSameRoleType = (
   return role.affiliation?.type === 'Organization' && role.affiliation?.id === newAffiliationId && isSameRoleType;
 };
 
-const findProjectManagerRoleThatHasAffiliation = (contributor: ProjectContributor, affiliationId: string) => {
-  return contributor.roles.findIndex((role) => role.affiliation?.id === affiliationId && isProjectManagerRole(role));
+const findProjectManagerRoleThatHasAffiliation = (roles: ProjectContributorRole[], affiliationId: string) => {
+  return roles.findIndex((role) => role.affiliation?.id === affiliationId && isProjectManagerRole(role));
 };
 
-const findNonProjectManagerRoleThatHasAffiliation = (contributor: ProjectContributor, affiliationId: string) => {
-  return contributor.roles.findIndex((role) => role.affiliation?.id === affiliationId && isNonProjectManagerRole(role));
+const findNonProjectManagerRoleThatHasAffiliation = (roles: ProjectContributorRole[], affiliationId: string) => {
+  return roles.findIndex((role) => role.affiliation?.id === affiliationId && isNonProjectManagerRole(role));
 };
 
 export const findRoleIndexForAffiliation = (
   asProjectManager: boolean,
-  contributor: ProjectContributor,
+  roles: ProjectContributorRole[],
   affiliationId: string
 ) => {
   return asProjectManager
-    ? findProjectManagerRoleThatHasAffiliation(contributor, affiliationId)
-    : findNonProjectManagerRoleThatHasAffiliation(contributor, affiliationId);
+    ? findProjectManagerRoleThatHasAffiliation(roles, affiliationId)
+    : findNonProjectManagerRoleThatHasAffiliation(roles, affiliationId);
 };
 
 export const notLastOfItsRoleType = (
-  contributor: ProjectContributor,
+  roles: ProjectContributorRole[],
   affiliationId: string,
   roleType: ProjectContributorType
 ) => {
-  const rolesWithoutAffiliationId = contributor.roles.filter((role) => role.affiliation?.id !== affiliationId);
+  const rolesWithoutAffiliationId = roles.filter((role) => role.affiliation?.id !== affiliationId);
   return rolesWithoutAffiliationId.some((role) => role.type === roleType);
 };
+
+export enum AffiliationErrors {
+  NO_AFFILIATION_ID,
+  CAN_ONLY_BE_ONE_PROJECT_MANAGER,
+  ADD_DUPLICATE_AFFILIATION,
+  NO_ROLE_TO_CHANGE,
+  NO_ROLE_TO_REMOVE,
+}
+
+export const addAffiliation = (
+  newAffiliationId: string,
+  contributorRoles: ProjectContributorRole[],
+  asProjectManager = false,
+  labels?: LanguageString
+): { newContributorRoles?: ProjectContributorRole[]; error?: AffiliationErrors } => {
+  if (!newAffiliationId) {
+    return {
+      newContributorRoles: contributorRoles,
+      error: AffiliationErrors.NO_AFFILIATION_ID,
+    };
+  }
+
+  // There can only be one project manager role
+  if (asProjectManager && contributorRoles.some((role) => isProjectManagerRole(role) && !hasEmptyAffiliation(role))) {
```

Comment:
Hvorfor er `!hasEmptyAffiliation(role)` relevant her?

Reply:
Når vi legger til en ProjectManager eller ProjectParticipant må de ha minst én rolle (som da inneholder dette rollenavnet). Hvis vi ikke velger affiliation setter vi affiliation = undefined, og i viewet vises det som at affiliation ikke er valgt:

![image](https://github.com/user-attachments/assets/59166749-6287-4b84-8b15-111bcc4b31e2)

I values ser det imidlertid slik ut:

![image](https://github.com/user-attachments/assets/b12ebccd-b5fe-4b31-a917-d37b6f2835a1)

Men en undefined affiliation betyr ikke at vi ikke senere kan legge til ProjectManager-affiliation, for hvis affiliation er undefined må denne kunne erstattes av en som er defined. Derfor sjekker vi dette slik at vi ikke avviser en tilknytning hvis alt som fins fra før er undefined.